### PR TITLE
[ASCII-1533] Add security-agent config by-source subcommand and api endpoint

### PR DESCRIPTION
**Pull request by jeremy-hanna**
### What does this PR do?

https://datadoghq.atlassian.net/browse/ASCII-1533

similar to the work done in https://github.com/DataDog/datadog-agent/pull/26129 and https://github.com/DataDog/datadog-agent/pull/26056

This PR adds a metadata payload for the security-agent process configuration by layers to send to the backend

### Describe how to test/QA your changes

* `diagnose show-metadata security-agent` works
* configuration from the security-agent process is accurate
* `security-agent.yaml` is correctly added to the flare

### CODE REVIEW

File: comp/metadata/securityagent/def/component.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+// Copyright 2024-present Datadog, Inc.
+
+// Package def is the metadata provider for security-agent process
+package def
+
+// team: agent-shared-components
+
+// Component is the component type.
+type Component interface {
+	GetAsJSON() ([]byte, error)
```
Comment (hush-hush) [inline comp/metadata/securityagent/def/component.go:13]: Why do we need this method ?

File: comp/metadata/securityagent/impl/security_agent.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+}
+
+// Requires defines the dependencies for the securityagent metadata component
+type Requires struct {
+	Log        log.Component
+	Config     config.Component
+	Serializer serializer.MetricSerializer
+	// We need the authtoken to be created so we requires the comp. It will be used by configFetcher.
+	AuthToken authtoken.Component
+	// SysProbeConfig optional.Option[sysprobeconfig.Component]
```
Comment (hush-hush) [inline comp/metadata/securityagent/impl/security_agent.go:76]: Should we remove this line ?

File: comp/metadata/securityagent/impl/security_agent.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	// TODO: ??? do we use this
+	if !sa.conf.GetBool("inventories_configuration_enabled") {
+		return metadata
+	}
+
+	// FIXME: we should use a subset of the config from the security-agent
+	if sa.conf == nil {
+		sa.log.Debugf("security-agent config not available: disabling securityagent metadata")
+		return metadata
+	}
```
Comment (hush-hush) [inline comp/metadata/securityagent/impl/security_agent.go:129]: When would `sa.conf` be nil ?

File: comp/metadata/systemprobe/def/component.go [hunk 4]
```diff
@@ -9,4 +9,6 @@ package def
 // team: agent-shared-components
 
 // Component is the component type.
-type Component interface{}
+type Component interface {
+	GetAsJSON() ([]byte, error)
```
Comment (hush-hush) [inline comp/metadata/systemprobe/def/component.go:13]: Why add this method ?

File: comp/metadata/securityagent/impl/security_agent.go [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+	// FIXME: we should use a subset of the config from the security-agent
+	rawLayers, err := fetchSecurityAgentConfigBySource(sa.conf)
+	if err != nil {
+		sa.log.Debugf("error fetching security-agent config layers: %s", err)
+		return metadata
+	}
+
+	configBySources := map[string]interface{}{}
+	fmt.Println("rawLayers")
```
Comment (hush-hush) [inline comp/metadata/securityagent/impl/security_agent.go:139]: Left over from debug I think

File: comp/metadata/securityagent/impl/security_agent.go (hunk 3 above)
Comment (jeremy-hanna) [reply comp/metadata/securityagent/impl/security_agent.go:129]: this was just left over from the `optional.Option[sysprobconf.Comp]` error handling that I was going to reuse if we needed to make a securityagent config component - I will remove this

File: comp/metadata/securityagent/impl/security_agent.go (hunk 5 above)
Comment (jeremy-hanna) [reply comp/metadata/securityagent/impl/security_agent.go]: fixed in https://github.com/DataDog/datadog-agent/pull/26160/commits/b75328d0788ef1aaf102214124717a445c403d81

File: comp/metadata/systemprobe/def/component.go (hunk 4 above)
Comment (jeremy-hanna) [reply comp/metadata/systemprobe/def/component.go]: removed in https://github.com/DataDog/datadog-agent/pull/26160/commits/b75328d0788ef1aaf102214124717a445c403d81

File: comp/metadata/securityagent/impl/security_agent.go (hunk 3 above)
Comment (jeremy-hanna) [reply comp/metadata/securityagent/impl/security_agent.go]: removed in https://github.com/DataDog/datadog-agent/pull/26160/commits/b75328d0788ef1aaf102214124717a445c403d81

File: comp/metadata/securityagent/impl/security_agent.go (hunk 2 above)
Comment (jeremy-hanna) [reply comp/metadata/securityagent/impl/security_agent.go]: removed in https://github.com/DataDog/datadog-agent/pull/26160/commits/b75328d0788ef1aaf102214124717a445c403d81

File: comp/metadata/securityagent/def/component.go (hunk 1 above)
Comment (jeremy-hanna) [reply comp/metadata/securityagent/def/component.go]: removed in https://github.com/DataDog/datadog-agent/pull/26160/commits/b75328d0788ef1aaf102214124717a445c403d81

**Review by paulcacheux — APPROVED** [review]
LGTM for security agent owned files

**Review by hush-hush — APPROVED** [review]
One last comment, feel free to merge after fixing

**Pull request merged**